core/src/main/java/io/substrait/dsl/SubstraitBuilder.java

```diff
@@ -165,6 +166,28 @@ private Join join(
         .build();
   }
 
+  public HashJoin hashJoin(
+      int[] leftKeys, int[] rightKeys, HashJoin.JoinType joinType, Rel left, Rel right) {
```

Comment:
I would avoid using bare arrays here as it makes this hard to use in my opinion. With something like:
```java
  public HashJoin hashJoin(
      List<Integer> leftKeys, List<Integer> rightKeys, HashJoin.JoinType joinType, Rel left, Rel right)
```
users can inline the key lists like
```java
b.hashJoin(
  Collections.EMPTY_LIST,
  Collections.EMPTY_LIST,
  HashJoin.JoinType.INNER, commonTable, commonTable))
```
or
```java
b.hashJoin(
  List.of(1,3),
  List.of(2,4),
  HashJoin.JoinType.INNER, commonTable, commonTable)
```
assuming they have access to newer versions of Java which _we_ don't within the core library, but we do with the isthmus side of things.

Reply:
Ah I was thinking a better way 😃
I will update.

Reply:
Updated the code.